CONTRIBUTING.md

```diff
@@ -1,5 +1,7 @@
 # Contributing to Hazelcast
 
+**NOTE**: It is recommended to use Node 12 and above for development.
```

Comment:
I am fine with that, but we will still have our tests running with Node.js 10, right?

Reply:
I don't know if it is possible, but one solution might be dividing the dev dependencies into two subgroups: One for testing, and one for the rest (things like linters etc). We can make sure that testing dependencies are Node.js 10 compatible, and be flexible on the rest

Reply:
Yes see my comment below for more.